apis/api.ts

```diff
@@ -0,0 +1,28 @@
+export async function getUser() {
+  const response = await fetch("https://bootcamp-api.codeit.kr/api/users/1");
+  const body = await response.json();
+  return body;
+}
+
+export async function getLink(folderId: string | null) {
+  if (!folderId) {
+    const response = await fetch("https://bootcamp-api.codeit.kr/api/users/1/links");
+    const body = await response.json();
+    return body;
+  }
+  const response = await fetch(`https://bootcamp-api.codeit.kr/api/users/1/links?folderId=${folderId}`);
+  const body = await response.json();
+  return body;
+}
+
+export async function getFolderList() {
+  const response = await fetch("https://bootcamp-api.codeit.kr/api/users/1/folders");
+  const body = await response.json();
+  return body;
+}
+
+export async function getShared() {
+  const response = await fetch("https://bootcamp-api.codeit.kr/api/sample/folder");
+  const body = await response.json();
+  return body;
+}
```

Comment:
```suggestion
import axios from 'axios';

const apiInstance = axios.create({
  baseURL: 'https://bootcamp-api.codeit.kr/api',
})

export async function getUser() {
  const { data } = await apiInstance.get('/users/1');
  return data;
}

export async function getLink(folderId: string | null) {
  const queryString = folderId ? `?folderId=${folderId}` : ''
  const { data } = await apiInstance.get(`/users/1/links${queryString}`);
  return data;
}

export async function getFolderList() {
  const { data } = await apiInstance.get('/users/1/folders');
  return data;
}

export async function getShared() {
  const { data } = await apiInstance.get('/sample/folder');
  return data;
}

```

axios 라이브러리를 사용하면 이렇게 관리할수 있습니다.

1. baseURL 기준으로 api를 카테고라이징 할 수 있다.
2. 기회가 된다면 [qs](https://www.npmjs.com/package/qs) 라이브러리도 한번 사용해보세요.